### PR TITLE
[Documentation] Update api product documentation 

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: docs/api/products.rst [hunk 1]
```diff
@@ -766,13 +573,232 @@ Example Response
     The images should be passed in array as an attribute (files) of request. See how it is done in Sylius
     `here <https://github.com/Sylius/Sylius/blob/master/tests/Controller/ProductApiTest.php>`_.
 
-Updating Product
-----------------
+Getting a Single Product
+------------------------
 
-You can request full or partial update of resource. For full product update, you should use PUT method.
+To retrieve the details of the product you will need to call the ``/api/v1/product/product_id`` endpoint with the ``GET`` method.
```
Comment (lchrusciel) [inline docs/api/products.rst]: `product_id` -> `id`

File: docs/api/products.rst [hunk 2]
```diff
@@ -766,13 +573,232 @@ Example Response
...
+                     }
+                 }
+             ]
+         }
+     }
+
+Updating a Product
+------------------
+
+To fully update a product you will need to call the ``/api/v1/products/product_id`` endpoint with ``PUT`` method.
```
Comment (lchrusciel) [inline docs/api/products.rst]: `product_id` -> `id`

File: docs/api/products.rst [hunk 3]
```diff
@@ -3,283 +3,77 @@ Products API
...
 +-----------------+----------------------------------------------------------------------------+
-| main_taxon      | The main taxon to whose product is assigned                                |
+| main_taxon      | The main taxon to whose the product is assigned                            |
 +-----------------+----------------------------------------------------------------------------+
 
 
 .. note::
 
-    Read more about `Products`__
+	Read more about :doc: `Product </components/Product/models>`
```
Comment (TheMadeleine) [inline docs/api/products.rst]: ```Read more about :doc: `Product model in the component docs</components/Product/models>`.```

File: docs/api/products.rst [hunk 4]
```diff
@@ -296,131 +90,134 @@ Definition
...
+		"product_taxons": [],
+		"reviews": [],
+		"average_rating": 0,
+		"images": [],
+		"_links": {
+			"self": {
+				"href": "/api/v1/products/61"
+			}
+		}
+	}
```
Comment (TheMadeleine) [inline docs/api/products.rst]: missing blank line.

File: docs/api/products.rst [hunk 5]
```diff
@@ -452,322 +249,551 @@ Example
...
+		"_links": {
+			"self": {
+				"href": "/api/v1/products/62"
+			}
+		}
+	}
+
+.. note::
+
+	The images should be passed in array as an attribute (files) of request. See how it is done in Sylius
```
Comment (TheMadeleine) [inline docs/api/products.rst]: `The images (files) should be passed in an array as an attribute of request. (...)`

File: docs/api/products.rst [hunk 6]
```diff
@@ -784,173 +810,179 @@ Definition
 Example
 ^^^^^^^
 
+ To full update the product with ``id = 3`` use the below method:
```
Comment (TheMadeleine) [inline docs/api/products.rst]: fully

File: docs/api/products.rst [hunk 7]
```diff
@@ -784,173 +810,179 @@ Definition
...
+Exemplary Response
+^^^^^^^^^^^^^^^^^^
 
 .. code-block:: text
 
-    STATUS: 204 No Content
+	STATUS: 204 No Content
 
-If you try to perform full product update without all required fields specified, you will receive a 400 error.
+If you try to perform full product update without all required fields specified, you will receive a ``400 Bad Request`` error.
```
Comment (TheMadeleine) [inline docs/api/products.rst]: **a** full product update

File: docs/api/products.rst [hunk 8]
```diff
@@ -784,173 +810,179 @@ Definition
...
++===============+================+======================================+
+| Authorization | header         | Token received during authentication |
++---------------+----------------+--------------------------------------+
+| id            | url attribute  | Id of requested resource             |
++---------------+----------------+--------------------------------------+
 
 Example
 ^^^^^^^
 
+To partial update the product with ``id = 3`` use the below method:
```
Comment (TheMadeleine) [inline docs/api/products.rst]: partially

File: docs/api/products.rst (hunk 4 above)
Comment (tuka217) [inline docs/api/products.rst]: Done

**Comment by pjedrzejewski** [comment]
Thanks Ania!

**Pull request merged**